lib/src/layouts/yaru_compact_layout.dart

```diff
@@ -35,33 +40,62 @@ class YaruCompactLayout extends StatefulWidget {
   final IndexedWidgetBuilder pageBuilder;
 
   /// The index of the initial page to show.
-  final int initialIndex;
+  final int? initialIndex;
 
   /// Called when the user selects a page.
   final ValueChanged<int>? onSelected;
 
+  /// An optional controller that can be used to navigate to a specific index.
+  final YaruPageController? controller;
+
   @override
   State<YaruCompactLayout> createState() => _YaruCompactLayoutState();
 }
 
 class _YaruCompactLayoutState extends State<YaruCompactLayout> {
   late int _index;
+  late final int _length;
 
-  late ScrollController _controller;
+  late final ScrollController _scrollController;
+  late final YaruPageController _pageController;
 
   @override
   void initState() {
-    _controller = ScrollController();
-    _index = widget.initialIndex;
     super.initState();
+    _length = widget.length ?? widget.controller!.length;
+    _scrollController = ScrollController();
+    _updatePageController();
+    _index = max(_pageController.initialIndex, widget.initialIndex ?? 0);
   }
 
   @override
   void dispose() {
-    _controller.dispose();
+    _scrollController.dispose();
+    _pageController.removeListener(_pageControllerCallback);
+    if (widget.controller == null) _pageController.dispose();
     super.dispose();
   }
 
+  @override
+  void didUpdateWidget(covariant YaruCompactLayout oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.controller != oldWidget.controller) {
+      oldWidget.controller?.removeListener(_pageControllerCallback);
+      _updatePageController();
+    }
+  }
+
+  void _updatePageController() {
+    _pageController = widget.controller ?? YaruPageController(length: _length);
```

Comment:
should it pass `widget.length` because `_length` may be the previous controller's length?

Reply:
:+1: